### PR TITLE
Improve formulation of the "Main Features" list in the API Reference

### PR DESCRIPTION
**Pull request by yvonnefroehlich**
**Description of proposed changes**

This PR aims to partly re-formulate the "Main Features" list at the beginning of the API Reference (please see https://www.pygmt.org/dev/api/index.html#main-features) to make it more consistent and nicer to read.
Happy to receive feedback whether this is a good chnage🙂!

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If wrapping a new module, open a 'Wrap new GMT module' issue and submit reasonably-sized PRs.
- [ ] If adding new functionality, add an example to docstrings or tutorials.
- [ ] Use underscores (not hyphens) in names of Python files and directories.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

**Review by seisman — APPROVED** [review]

File: pygmt/__init__.py [hunk 1]
```diff
@@ -1,6 +1,6 @@
 """
 PyGMT is a library for processing geospatial and geophysical data and making
-publication quality maps and figures. It provides a Pythonic interface for the
+publication-quality maps and figures. It provides a Pythonic interface for the
 Generic Mapping Tools (GMT), a command-line program widely used in the Earth
 Sciences. Besides making GMT more accessible to new users, PyGMT aims to
```
[the remark this answers is not in the document]
Comment (michaelgrund) [reply pygmt/__init__.py:5]: In the course of these changes I think we can expand the description a little bit based on the intro on the [GMT main site](https://www.generic-mapping-tools.org/).

[the remark this answers is not in the document]
Comment (yvonnefroehlich) [reply pygmt/__init__.py:5]: I am fine with this addition. Should we also update the text in the overview section? Please see https://github.com/GenericMappingTools/pygmt/blame/main/doc/overview.rst the lines 4-11:
```
About
-----

PyGMT is a Python wrapper for the
`Generic Mapping Tools (GMT) <https://github.com/GenericMappingTools/gmt>`__, a
command-line program widely used in the Earth Sciences.
It provides capabilities for processing spatial data (gridding, filtering, masking,
FFTs, etc) and making high quality plots and maps.
```

[the remark this answers is not in the document]
Comment (yvonnefroehlich) [reply pygmt/__init__.py]: These two files were updated in commit https://github.com/GenericMappingTools/pygmt/pull/2336/commits/9b26142301af82b065b7ebbd62e9d7c5174d5657.

[the remark this answers is not in the document]
Comment (yvonnefroehlich) [reply pygmt/__init__.py]: Also updated the overall README in commit https://github.com/GenericMappingTools/pygmt/pull/2336/commits/f92015d0c0256c957ac7e1cb466d561defb1e9e2.

**Review by michaelgrund — APPROVED** [review]

**Pull request merged**